langchain/src/smith/runner_utils.ts

```diff
@@ -59,6 +59,30 @@ class RunIdExtractor {
   }
 }
 
+class SingleRunExtractor extends BaseTracer {
+  runPromiseResolver: (run: Run) => void;
+
+  runPromise: Promise<Run>;
+
+  /** The name of the callback handler. */
+  name = "single_run_extractor";
+
+  constructor() {
+    super();
+    this.runPromise = new Promise<Run>((extract) => {
+      this.runPromiseResolver = extract;
+    });
+  }
+
+  async persistRun(run: Run) {
+    this.runPromiseResolver(run);
```

Comment:
does this always work for nested runs? Or do we need to check that it's the root trace?

Reply:
Not sure